zebra-rpc/src/methods.rs

```diff
@@ -190,11 +193,87 @@ where
     }
 
     fn get_blockchain_info(&self) -> Result<GetBlockChainInfo> {
-        // TODO: dummy output data, fix in the context of #3143
-        //       use self.latest_chain_tip.estimate_network_chain_tip_height()
-        //       to estimate the current block height on the network
+        let network = self.network;
+
+        // `chain` field
+        let chain = self.network.bip70_network_name();
+
+        // `blocks` field
+        let tip_height = self
+            .latest_chain_tip
+            .best_tip_height()
+            .ok_or_else(|| Error {
+                code: ErrorCode::ServerError(0),
+                message: "No Chain tip available yet".to_string(),
+                data: None,
+            })?;
+
+        // `best_block_hash` field
+        let tip_hash = self.latest_chain_tip.best_tip_hash().ok_or_else(|| Error {
+            code: ErrorCode::ServerError(0),
+            message: "No Chain tip available yet".to_string(),
+            data: None,
+        })?;
+
+        // `estimated_height` field
+        let estimated_height = self
```

Comment:
Yea, you are right, i missed that. 

I was wondering how to do this. Can we just compare? Something as:

```
if tip_height > estimated_height {
   estimated_height = tip_height;
}
else {
   estimated_height = estimated_height;
}
```

Or do we need to go deeper and create an additional method in the chaintip ?

By the way, looking at `estimate_network_chain_tip_height()` definition i don't know why it returns an `Option` if it always return `Some()`. https://github.com/ZcashFoundation/zebra/blob/main/zebra-chain/src/chain_tip.rs#L46-L57
So another option may be to return `None` from there if we are already in the tip (i am not sure how hard will be to implement that)

Reply:
The `?` operator on this line returns `None` if `best_tip_height_and_block_time()` returns `None`.
https://github.com/ZcashFoundation/zebra/blob/5d7f9861830ce4f5de9d5d68df15854a56610f55/zebra-chain/src/chain_tip.rs#L51

Reply:
> Or do we need to go deeper and create an additional method in the chaintip ?

### Priority

I can't see where this field is used in lightwalletd or ZecWallet Lite. So we might just be able to open a ticket and fix it later.

I've just sent an email to Aditya to double-check.

### Specification

I checked what `zcashd` does here:
1. If it has stopped syncing, return the current height: https://github.com/zcash/zcash/blob/6cd5b8792bbe0a3cc1064e1884914415f0c6e7d7/src/rpc/blockchain.cpp#L1068-L1071
2. If the block time is ahead of the current time, return the current height: https://github.com/zcash/zcash/blob/f7006e9a33ca37ae88520e434ee5c2b2498c1cc6/src/metrics.cpp#L141-L143
3. Otherwise, estimate the tip height: https://github.com/zcash/zcash/blob/f7006e9a33ca37ae88520e434ee5c2b2498c1cc6/src/metrics.cpp#L145-L160

Here's the `SyncStatus` method that returns if we're close to the tip:
https://github.com/ZcashFoundation/zebra/blob/5d7f9861830ce4f5de9d5d68df15854a56610f55/zebrad/src/components/sync/status.rs#L46-L47

### Zebra Changes

Here's one way to do this fix:
1. Move `SyncStatus` to `zebra_node_services`
2. Add `SyncStatus` to `RpcImpl`, and use it to check if we're close to the tip
3. Fix `NetworkChainTipHeightEstimator` to return the current height if the block time is ahead of the current time

Here's the `NetworkChainTipHeightEstimator` code. Removing negative times should simplify it a lot. The bug fix will probably make some tests fail, Janito can help you fix them.

https://github.com/ZcashFoundation/zebra/blob/main/zebra-chain/src/chain_tip/network_chain_tip_height_estimator.rs#L111-L120

Reply:
Aditya said a rough estimate was ok.

So let's go with a simple fix for now. I will open a ticket for the rest.

What do you think of these changes?
1. If the block time is ahead of the current time, return the current height
2. If the estimate is below the current height, return the current height

Reply:
I put the rest of the changes in #3954

Reply:
Added the simple fix in https://github.com/ZcashFoundation/zebra/pull/3891/commits/56f6ba37ad350ca35b3208139ef0b3e9a94ff59a